usagov/usagov-2021
# Usagov 439 import federal directory data

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-439

## Description

This is an update to the configuration of the Feed structure for import of Federal Agency records, and to the agency_import_prep.php script that combines and transforms exports from mothership for import. The "Synonym" field is also removed from the Federal Directory Record content type. 

### How to test

Download the files at  https://drive.google.com/drive/folders/1xw33rd_TGSGA9ExF1QX2EOjnzFWMMEgx?usp=sharing
These are exports of Directory Records from mothership as of 2022-10-19. 

You can follow these instructions (added as part of this PR) to do the import -- just skip the section about exporting from mothership: 

https://github.com/usagov/usagov-2021/blob/USAGOV-439-import_federal_directory_data/web/modules/custom/usagov_directories/docs/Importing_Federal_Agency_Records.md

I recommend importing the first file, as documented, and one additional file to test the instructions. You should wind up with a good selection of Federal Directory Records. 

**Out of scope for this ticket:** Synonyms, English/Spanish toggles. Those are on separate Jira tickets. 

### Changes in more detail

Feed structure changes: 
* Removed the synonym field from the importer
* Changed language code to zxx ("Not Applicable") -- reason is that the feed contains both English and Spanish items. Unfortunately I still sometime need to set the language when an import updates existing items. 
* Set the owner of new items to user 19 (Shoshana) 
* Add second and third Phone, TTY,  and Toll numbers to the mappings
* Add Street 3 to mappings
(Mappings are the base fields that can be set for every feed. It's still necessary to add mappings when a feed includes multiple links for any of the link types.) 

Transformation script changes: 
* No longer splits transformed CSV files based on how many of each phone type field are populated -- turned out not to be necessary. 
* If Subdivision field is populated in the mothership export, sets Street 1 to Subdivision value and moves Street 1 and Street 2 values into Street 2 and Street 3

Federal Directory Record change: 
* Removed the Synonym field.

docs/Importing_Federal_Agency_Records.md: new documentation file. 

## Checklist for the Developer


- [ ] The JIRA ticket contains clear testing steps. (See above instead.) 
- [] Any preparation/installation/update steps required for this code to work properly (drush commands, scripts, configuration updates) are documented in the ticket.

## Checklist for the Peer Reviewers
